### DISCUSSION TITLE
New to this all, and would like some help

### DISCUSSION THREAD

**Discussion by budderflymoo** [isid0re/SoloLeveling#49] [Q&A] [answered]
Am I to understand, That I need to install/paste the sololeveling zip file....
and then go on here and update all of the scripts that have been posted? 

I have been having issues since yesterday. I had a sorc upto 73, and now she wont load. 
Im getting syntax errors, currently its the sololeveling.js line 106 that is an issue... 
Replacing the JS file did not fix this error... do i just keep going?

**Comment by isid0re** [comment]
I'm not sure which file you are upgrading from. If it was one of the original versions. Where all the scripts and functions were located in 1 file then you will need to update everything. 

When you download and extract the zip file it will make 1 folder with 3 folders inside. Config bots and sololeveling
The easiest install method is to just drag (or copy and paste which ever you prefer) the 3 folders into the kolbot/libs folder.  It will then ask you to replace existing files. Say yes.  

After you are done installing you will have a libs folder that should match the installation instructions from the readme.  Let me know if you are still having any problems.

**Reply by budderflymoo** [reply]
Hi There, thank you so much for your response. 
I had a character working very well, and then i updated a couple days ago and now I get errors all over the place.

I noticed that there are edit files listed here, and are not in the original zip. So what i did was watch the error codes come up in the log screen and replaced the .js files that were coming up bad.  My original question i guess would be : If i download the sololeveling zip file and install, i shouldnt need to edit or update, as it should have all current files?

I appear to have a townoverride error now. ( kw is undefined, and..... bloodraven are 2 that I recall having issues with also)

My new guy is running and cant seem to get anywhere, it does the countess and then times out due to wp i think. 

I am very new to all of this, and find it all very interesting. I apologize if my questions are dumb. I included the area in the JS that is problematic.... ive tried to close and include the "return true" into the syntax thing, but I obviusly know very little lol.

		if (sor.interact()) {
			print('ÿc9SoloLevelingÿc0: used scroll of resistance');
		} else {
			clickItem(1, sor);
			print('ÿc9SoloLevelingÿc0: used scroll of resistance');
		
		return true;
		}
		};

**Comment by isid0re** [comment]
I appreciate your thoroughness =) so a couple of things. I have been working thru improving various sequences and that led to errors. I've just made a push that fixes those errors improves the quest sequencing and gear handling. It took a couple of days to iron it out. So what most likely happened is your download had those issues.

Please delete the old files and make a fresh install from the repo as of this post. I have already updated the repo with the fixes so your install shouldnt have any more problems. 

Please let me know how it goes. Thanks

**Reply by budderflymoo** [reply]
Yeah, everthing went sideways in the last 2 days.
I will reinstall and try again. Ive done that 3 times now, i am a pro..... Can I not just "update" like I do the kolbot files? 
Or I guess it isnt linked into the update thingy.... (sorry my terms are not all there)

I have learned so much already by just finding the files, lines, and trying to correct them on my own. so its been run. 
One more thing: 

In solo leveling, it says that I cannot run boss area scripts at the same time. why? 
and to further this, if i turn it on, with solo, it still works.... im guessing it will create errors as it goes because the paths conflict?
Just curious. 

Now lets see if redownloading fixes this pit issue!
Coffees on me!

**Comment by isid0re** [comment]
So you can run boss areas. I just suggest you don't. My reasoning is, these scripts are run after sololeveling. And many of those scripts would just be repeated during the sololeveling sequence.

If you use tortoise svn, and add sololeveling to your kolbot directory. You should be able to update like you would with standard kolbot files. I've not done that but I'm sure some people have. Hopefully someone else can reply here and help you there.

**Reply by budderflymoo** [reply]
It appears that If  turn off boss scripts off, it wont run, just sits in town. 
I dont know whats wrong with her right now, but its the sololeveling bot file im guessing? She sits in the tunnels and tries to find the way point to go to the pits, but she errors out, as she doesnt go to town first, which i assume she is supposed to do. 

Error in SoloLeveling (pather.js #1021) useWaypoint: Failed to use waypoint (Area: 19, ....
	pather.js line1021 is :	throw new Error("useWaypoint: Failed to use waypoint");

I redid the files. I copied my character files prior to redoing them, and pasted them back into the config file....

**Comment by isid0re** [comment, marked as answer]
Try using the redownload config file. And rename it to your character file instead of using the old config file. And leave the settings stock.

**Comment by budderflymoo** [comment]
Thank you for your help, she is running now... although obsessed with a crap amulet....

If i left everything as is, and never updated any of the files would it continue to run as is?

**Comment by isid0re** [comment]
Yes it will continue to run as is. Also as a helpful tip. 

If you see I update the files Click on the commits button. There is a description of what changes occured in the files. 

I'm actively working out any errors and omissions as I improve the scripts and performance. So if you find any errors or see possible improvements as you watch the bot run, please post as an issue so I can keep a tracking record on it. Thanks.

**Reply by budderflymoo** [reply]
Good morning, Just updated the new file you had uploaded. Can I simply just copy paste the files that you updated? or does it all need to install together for some reason? 

Also, I was looking at the code, specifically the pits code

function pits () {
-	if (me.diff !== 2 && goldCheck()) {
+	if (me.diff === 0 && !Pather.accessToAct(4) || me.diff !== 2 && goldCheck()) {
		return true;
		
Does this mean, that it will not go to the pits unless it now has access to act 4?

Thanks! Love it!

**Reply by isid0re** [reply]
Correct in a normal difficulty game it wont enter pits unless it has access to act 4 hell. At which point it will use the gold check function. 

Yes you can just copy in the modified files instead of everything.
